service/history/tasks/task.go

```diff
@@ -54,6 +55,14 @@ type (
 	HasDestination interface {
 		GetDestination() string
 	}
+
+	// PartialTask is a task that is generated without a workflow key or version.
+	// The missing fields are meant to be filled in by mutable state.
+	PartialTask interface {
```

Comment:
I feel this definition should live in the new statemachine package. or the same place where the Environment interface is implemented.

Reply:
Yeah, that makes sense.